packages/app/src/cli/services/function/common.ts

```diff
@@ -40,11 +40,12 @@ export async function inFunctionContext(
 
 export async function ensureFunctionExtensionFlavorExists(
   specification: FunctionSpec,
-  extensionFlavor: ExtensionFlavor,
+  extensionFlavor: ExtensionFlavorValue,
   templateFlavor: TemplateFlavor,
   templateDownloadDir: string,
 ): Promise<string> {
-  const templatePath = specification.templatePath(templateFlavor)
+  // Extensions resolves the template path from template flavor, but templates use extension flavor
```

Comment:
wat. Can we simplify this to use just one flavor? feels hard to understand what's happening here 😅

Reply:
the problem is that the resolution of the `templatePath` is different between how current local specifications behave and the one we load from the template specificacions:
- Local specs: Replace the token in a template url with the flavor
- Template specs: Each flavor includes it's own template url
Once we use only the remote templates this should be changed to:

```ts
const templatePath = specification.templatePath(extensionFlavor)
```

Reply:
Ah well, since we are removing the local specs, we can simplify this then, cool 👌